dependencies.yaml

```diff
@@ -611,6 +612,14 @@ dependencies:
           - pytest
           - pytest-cov
           - pytest-xdist
+    specific:
+      - output_types: [conda, requirements, pyproject]
+        matrices:
+          - matrix: {cuda: "12.*"}
+            packages:
+              - pynvjitlink
+          - matrix:
+            packages:
```

Comment:
This isn't the right fix. This adds pynvjitlink as a testing dependency, rather than as a package dependency. We need this to appear in cudf's `pyproject.toml` for CUDA 12. Unfortunately, we don't support matrix entries for pyproject.toml yet (https://github.com/rapidsai/dependency-file-generator/issues/46). Therefore, the proper fix is to modify `ci/build_wheel.sh` and change one of the lines that deletes ptxcompiler to be a find-replace ptxcompiler with pynvjitlink. I will push a change for that.

Reply:
I see, thanks for the context @bdice !

Reply:
Fixed in 8b8a12c2ba3b75008645090c3db856bb1b7518d9.